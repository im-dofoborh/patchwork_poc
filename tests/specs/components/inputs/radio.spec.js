import { mount, createLocalVue } from '@vue/test-utils';
import RadioInput from '~/components/inputs/Radio';

describe('RadioInput component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('radio-input', RadioInput);
    wrapper = mount(RadioInput, {
      localVue,
      propsData: {
        label: 'foo',
        name: 'bar',
        id: 'baz',
      }
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('emits input event on click', () => {
    const input = wrapper.find('input');
    input.trigger('click');
    expect(wrapper.emitted().input).toBeTruthy();
  });
});
