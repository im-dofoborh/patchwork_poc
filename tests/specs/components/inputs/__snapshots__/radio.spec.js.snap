// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioInput component renders default variant 1`] = `
<div class="sg-input-radio input-radio"><input id="baz" name="bar" type="radio" class="input-radio__input" value="foo"> <label for="baz" class="input-radio__label js-supress-focus-on-click">
    foo
  </label></div>
`;
